lib/web/apiserver_test.go

```diff
@@ -978,109 +978,115 @@ func TestResolveServerHostPort(t *testing.T) {
-		authProvider AuthProvider
-		expectedErr  string
-	}{
-		{
-			expectedErr:  "invalid session",
-			authProvider: makeProvider(validNode),
-			req: TerminalRequest{
-				SessionID: "",
-				Login:     validLogin,
-				Server:    validServer,
-				Term:      validParams,
+		}, {
+			expectedErr: "server: missing server",
+			cfg: TerminalHandlerConfig{
+				sessionData: session.Session{
+					ID:       session.NewID(),
+					Login:    "root",
+					ServerID: "",
+				},
 			},
 		},
 		{
-			expectedErr:  "bad term dimensions",
-			authProvider: makeProvider(validNode),
-			req: TerminalRequest{
-				SessionID: validSID,
-				Login:     validLogin,
-				Server:    validServer,
-				Term: session.TerminalParams{
-					H: -1,
-					W: 0,
+			expectedErr: "term: bad dimensions(-1x0)",
+			cfg: TerminalHandlerConfig{
+				sessionData: session.Session{
+					ID:       session.NewID(),
+					Login:    "root",
+					ServerID: uuid.New().String(),
+				},
+				term: session.TerminalParams{
+					W: -1,
+					H: 0,
 				},
 			},
 		},
 		{
-			expectedErr:  "invalid server name",
-			authProvider: makeProvider(validNode),
-			req: TerminalRequest{
-				Server:    "localhost:port",
-				SessionID: validSID,
-				Login:     validLogin,
-				Term:      validParams,
+			expectedErr: "term: bad dimensions(1x4097)",
+			cfg: TerminalHandlerConfig{
+				sessionData: session.Session{
+					ID:       session.NewID(),
+					Login:    "root",
+					ServerID: uuid.New().String(),
+				},
+				term: session.TerminalParams{
+					W: 1,
+					H: 4097,
+				},
 			},
 		},
 	}
 
-	ctx := context.Background()
-	for _, testCase := range validCases {
-		term, err := NewTerminal(ctx, testCase.req, testCase.authProvider, nil)
-		require.NoError(t, err)
-		require.Empty(t, cmp.Diff(testCase.req, term.params))
-		require.Equal(t, testCase.expectedHost, testCase.expectedHost)
-		require.Equal(t, testCase.expectedPort, testCase.expectedPort)
-	}
-
 	for _, testCase := range invalidCases {
-		_, err := NewTerminal(ctx, testCase.req, testCase.authProvider, nil)
-		require.Regexp(t, ".*"+testCase.expectedErr+".*", err.Error())
+		_, err := NewTerminal(ctx, testCase.cfg)
+		require.Equal(t, err.Error(), testCase.expectedErr)
 	}
+
+	validNode := types.ServerV2{}
+	validNode.SetName("eca53e45-86a9-11e7-a893-0242ac0a0101")
+	validNode.Spec.Hostname = "nodehostname"
+
+	// Valid Case
+	validCfg := TerminalHandlerConfig{
+		term: session.TerminalParams{
+			W: 100,
+			H: 100,
+		},
+		sctx: &SessionContext{},
+		authProvider: authProviderMock{
+			server: validNode,
+		},
+		sessionData: session.Session{
+			ID:       session.NewID(),
+			Login:    "root",
+			ServerID: uuid.New().String(),
+		},
+		keepAliveInterval:  time.Duration(100),
+		proxyHostPort:      "1234",
+		interactiveCommand: make([]string, 1),
+		displayLogin:       "tree",
+	}
+
+	term, err := NewTerminal(ctx, validCfg)
+	require.NoError(t, err)
+	// passed through
+	require.Equal(t, term.ctx, validCfg.sctx)
```

Comment:
nit: the order of your assertions here are all backwards. This makes for a confusing error message if one fails.

The syntax is `require.Equal(t, EXPECTED, ACTUAL)`

Reply:
Heh woops, I guess I'm used to an alternative assertion lib :D